src/data/set/intervals/instances.lean

```diff
@@ -0,0 +1,82 @@
+/-
+Copyright (c) 2022 Stuart Presnell. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Stuart Presnell
+-/
+import data.set.intervals.basic
+import algebra.order.ring
+import algebra.group_power.order
+
+/-!
+# Structure instances for unit intervals
+
+For suitably structured underlying type `α`, we exhibit the structure of
+the unit intervals from `0` to `1`.
+-/
+
+variables {α : Type*} [ordered_semiring α]
+
+open set
+
+/-- Instances for `(Icc 0 1)` -/
+
+instance : has_zero (Icc (0:α) 1) := { zero := ⟨0, left_mem_Icc.2 zero_le_one⟩ }
```

Comment:
For each of these you should add a `coe_*` lemma saying ↑(1 : Icc _ _) = (1 : a)`